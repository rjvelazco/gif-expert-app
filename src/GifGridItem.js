import React from 'react'

export const GifGridItem = ({title, url}) => {
    console.table({title, url});
    return (
        <div>
            <img src={url} alt={title} />
            <p>{ title }</p>
        </div>
    )
}
