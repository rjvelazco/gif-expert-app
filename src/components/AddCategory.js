import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const input = useRef();
    const handleInputChange = ({ target }) => {
        setInputValue(target.value);
        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(inputValue);
        if(inputValue.trim().length) {
            setCategories((categories) => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    // Manage input here to avoid fail the test
    // because we are not simulation the DOM on tests in this section
    // useEffect(()=> {
    //     if( !inputValue.length ) {
    //         input.current.value='';
    //     }
    // }, [inputValue])

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input
                ref={input}
                type="text"
                placeholder="Add Category"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};