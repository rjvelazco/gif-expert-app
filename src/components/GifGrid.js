import React from 'react'

export const GifGrid = ({ category }) => {
    const api_key = 'GyTgDRvJi4SIjxKQSRDSKA5CNussIOTV';
    const getGifs = async () => {

        const url = `https://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=${api_key}`;
        const resp = await fetch( url);
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })

        console.log(gifs);

    }
    getGifs();
    return (
        <div>
            <h3>{ category }</h3>
        </div>
    )
}
