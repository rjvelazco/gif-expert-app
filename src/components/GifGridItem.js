import React from 'react'

export const GifGridItem = ({category, title, url}) => {
    const repTitle =  `${category } - gif`.toUpperCase();
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{ title ? title : repTitle }</p>
        </div>
    )
}
