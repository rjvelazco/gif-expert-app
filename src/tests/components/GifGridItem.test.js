import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Tests on <GifGridItem />', () => {

    const title = 'Title';
    const url = 'https://localhost/algo.jpg';
    const category = 'Anime';
    const wrapper = shallow(
        <GifGridItem title={title} url={url} category={category} />
    );


    test('It should render GrifGridItem Component Correctly', () => {

        // wrapper.setProps({ title: 'setProps Methods' });
        // wrapper.update();
        expect(wrapper).toMatchSnapshot();

    });

    test('It should have a <p> with the title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    });

    test('It should have a <img> with the url and alt props value', () => {
        const img = wrapper.find('img');
        // console.log( img.props() );
        // img.props() -> Todos los props;
        // img.prop('porpName') -> Solo el prop especificado.
        expect( img.props().src ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    });

    test('It should have a <div> with the class "card"', () => {
        const div = wrapper.find('div');
        // console.log(div.props());
        expect( div.prop('className') ).toContain('card');
    })

});
