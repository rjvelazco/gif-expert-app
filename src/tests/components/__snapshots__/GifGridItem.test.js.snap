// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> It should render GrifGridItem Component Correctly 1`] = `
<div
  className="card"
>
  <img
    alt="Title"
    src="https://localhost/algo.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
